### PR TITLE
Gather post-install logic inside manifest

### PR DESCRIPTION
**Pull request by ghisvail**

### CODE REVIEW

**Comment by ghisvail** [comment]
Fails appdata validation `<release> description should be prose and not contain hyperlinks`. This is why integrating upstream metainfo can be challenging at times.

**Comment by ghisvail** [comment]
Advice from Flatpak IRC channel: relax validation with `skip-appstream-check`.

**Comment by ghisvail** [comment]
cc @ntninja

File: fixup-exports.py [hunk 1]
```diff
@@ -1,160 +0,0 @@
...
-		app_id = appid_elem.text
-		if app_id != to_name:
-			extra_provides.add(app_id)
-			appid_elem.text = app_id = to_name
-	
-	# Update desktop file location to match new appid/`to_name`
-	for launchable_elem in appdata.findall("./launchable[@type='desktop-id']"):
-		desktop_id = launchable_elem.text
-		if desktop_id.endswith(".desktop"):
-			desktop_id = desktop_id[:-len(".desktop")]
```
Comment (ntninja) [inline fixup-exports.py:90]: This needs to be incorporated or the app will not be launchable from software managers after installation. (See other comment.)

**Review by ntninja — CHANGES_REQUESTED** [review]
Your right, this isn't too bad after all! :+1:

Added some comments pointing out a couple of things that you forgot in this conversion.

File: fixup-exports.py (hunk 1 above)
Comment (ghisvail) [reply fixup-exports.py:90]: Done with `appstream-util modify launchable`

File: fixup-exports.py [hunk 2]
```diff
@@ -1,160 +0,0 @@
...
-	kill_list |= rename_appdata_with_desktop(args.prefix, "calibre-ebook-viewer", f"{args.appid}.ebook_viewer")
-	kill_list |= rename_desktop_with_icon(args.prefix, "calibre-lrfviewer", f"{args.appid}.lrfviewer")
-	
-	# Remove now unused files
-	for path in kill_list:
-		path.unlink()
-	
-	# Declared MIME types
-	xdg_mime_package_dir = args.prefix / "share" / "mime" / "packages"
-	shutil.move(xdg_mime_package_dir / "calibre-mimetypes.xml", xdg_mime_package_dir / f"{args.appid}.xml")
```
[the remark this answers is not in the document]
Comment (ghisvail) [reply fixup-exports.py:154]: Thank you for spotting it :+1:

File: com.calibre_ebook.calibre.yaml [hunk 3]
```diff
@@ -38,9 +38,30 @@ modules:
...
+      - desktop-file-edit --set-key Icon --set-value $FLATPAK_ID.ebook-viewer $FLATPAK_DEST/share/applications/$FLATPAK_ID.lrfviewer.desktop
+
+      # post-install for calibre
+      - mv $FLATPAK_DEST/share/applications/{calibre-gui,$FLATPAK_ID}.desktop
+      - for p in 16 32 48 64 128 256; do mv $FLATPAK_DEST/share/icons/hicolor/${p}x${p}/apps/{calibre-gui,$FLATPAK_ID}.png; done
+      - desktop-file-edit --set-key Icon --set-value $FLATPAK_ID $FLATPAK_DEST/share/applications/$FLATPAK_ID.desktop
+
+      # post-install for metainfo
+      - patch -i metainfo.patch -u $FLATPAK_DEST/share/metainfo/calibre-gui.metainfo.xml
+      - mv $FLATPAK_DEST/share/metainfo/{calibre-gui,$FLATPAK_ID}.metainfo.xml
```
[the remark this answers is not in the document]
Comment (ghisvail) [reply com.calibre_ebook.calibre.yaml]: Done

**Review by ghisvail — COMMENTED** [review]
@ntninja can you double-check please?

**Review by ntninja — APPROVED** [review]
Cross-referenced everything from the old script again, looking good now!

**Pull request merged**